### ISSUE TITLE
[TypeScript] [All] ts编译非常慢

### ISSUE THREAD

**Issue by BTBMan** [element-plus/element-plus#8889]
Bug Type: **`TypeScript`**

## Environment

- Vue Version: `3.2.25`
- Element Plus Version: `2.2.9`
- Browser / OS: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:102.0) Gecko/20100101 Firefox/102.0`
- Build Tool: `Vite`

### Related Component

- `All`

### Reproduction Link

[Github Repo](https://github.com/BTBMan/issue-demos)

### Steps to reproduce

```bash
pnpm i
pnpm run compileElement # 编译非常慢 大概20多秒
pnpm run compileAnt # 对比ant的编译
```

### What is Expected?

编译速度不应过慢

### What is actually happening?

编译速度很慢

### Additional comments

_(empty)_

**Comment by BTBMan** [comment]
发现了不止是打包慢 vscode的ts-server扫描也非常慢

**Comment by sxzz** [comment]
TypeScript 的性能问题暂时无解。如果不需要生成 dts 建议使用 esbuild、swc、babel 等类型擦除工具。

**Comment by BTBMan** [comment]
OK, Thanks for reply.

**Comment by BTBMan** [comment]
Did you find the reason for this?

**Comment by BTBMan** [comment]
More than 900 declaration files were generated, Can we just generate the most used declaration file such as components? I think it can improve speed.

**Comment by sxzz** [comment]
I don't think so. dts of Components depend on other libs (utils, constants...)

**Comment by BTBMan** [comment]
Yes, It's a hot potato.
